packages/netlify-cms-widget-markdown/src/serializers/index.js

```diff
@@ -135,7 +135,8 @@ export const remarkToMarkdown = obj => {
     .use(remarkToMarkdownPlugin, remarkToMarkdownPluginOpts)
     .use(remarkAllowAllText)
     .use(createRemarkShortcodeStringifier({ plugins: getEditorComponents() }))
-    .stringify(processedMdast);
+    .stringify(processedMdast)
+    .replace(/\r?/g, '');
```

Comment:
I think you should use the exact regexp I used so that any weird line endings (`\r\n`, `\n\r`, `\r`) all get normalized, not just the two first ones.

```suggestion
    .replace(/\r\n?|\n\r?/g, '\n');
```

Reply:
@papandreou the reason why we aren't using the exact regexp is because the problem we are seeing is not limited to line endings.

We're seeing CRs at the start of paragraphs, in the middle of paragraphs and not always with a LF next to them. On that basis, we felt it was best to just strip them out.

Reply:
Eeeew, yeah, that looks really messed up 🤮
I wonder if those carriage returns (before `In this blog,` and `**Can we make AI super?**`) weren't even shown in the markdown editor. If they weren't, I guess there might also be bugs in Slate's support for "alternative" line endings.

Either way I agree that this would be a pragmatic fix. There probably aren't any relevant platforms left that use `\r` line endings.

Reply:
@papandreou , from my comments on #1437 :
```
linaro.org_website on  remotes/origin/cms/homepage~2 [!] 
➜ hexdump -C _posts/events/2019-11-13-webinar-running-a-fuzzing-tool-in-op-tee.md | grep -E -C1 'Abstract|Martijn' | grep -C1 ' 0d '
000003b0  33 32 30 33 38 38 0d 0a  0a 2a 2a 2a 2a 0a 0a 2a  |320388...****..*|
000003c0  2a 41 62 73 74 72 61 63  74 2a 2a 3a 0d 20 0a 0a  |*Abstract**:. ..|
000003d0  54 72 75 73 74 65 64 20  45 78 65 63 75 74 69 6f  |Trusted Executio|
--
00000630  66 69 63 61 6c 6c 79 20  61 74 20 4f 50 2d 54 45  |fically at OP-TE|
00000640  45 2e 0d 0a 0a 0d 4d 61  72 74 69 6a 6e 20 77 69  |E.....Martijn wi|
00000650  6c 6c 20 64 69 73 63 75  73 73 20 74 68 65 20 63  |ll discuss the c|

linaro.org_website on  remotes/origin/cms/homepage~2 [!] 
➜
```
Not what I expected to see! It's also obvious in [PR 928](https://github.com/Linaro/website/pull/928/commits) to one of our websites.

Reply:
@papandreou , you can see this issue on my comment on #1437 :

```
linaro.org_website on  remotes/origin/cms/homepage~2 [!] 
➜ hexdump -C _posts/events/2019-11-13-webinar-running-a-fuzzing-tool-in-op-tee.md | grep -E -C1 'Abstract|Martijn' | grep -C1 ' 0d '
000003b0  33 32 30 33 38 38 0d 0a  0a 2a 2a 2a 2a 0a 0a 2a  |320388...****..*|
000003c0  2a 41 62 73 74 72 61 63  74 2a 2a 3a 0d 20 0a 0a  |*Abstract**:. ..|
000003d0  54 72 75 73 74 65 64 20  45 78 65 63 75 74 69 6f  |Trusted Executio|
--
00000630  66 69 63 61 6c 6c 79 20  61 74 20 4f 50 2d 54 45  |fically at OP-TE|
00000640  45 2e 0d 0a 0a 0d 4d 61  72 74 69 6a 6e 20 77 69  |E.....Martijn wi|
00000650  6c 6c 20 64 69 73 63 75  73 73 20 74 68 65 20 63  |ll discuss the c|

linaro.org_website on  remotes/origin/cms/homepage~2 [!] 
➜
```

and on a recent PR to one of our websites [# 928](https://github.com/Linaro/website/pull/928/commits)